### PR TITLE
[1.20.x] The 1.20 Update

### PR DESCRIPTION
**Pull request by ChampionAsh5357**
Oh hey, it's an update of the docs to 1.20 before 1.20 is released. Will remain a draft until 1.20 is released.

### CODE REVIEW

File: docs/concepts/registries.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ An example of a mod registering a custom block:
 ```java
 private static final DeferredRegister<Block> BLOCKS = DeferredRegister.create(ForgeRegistries.BLOCKS, MODID);
 
-public static final RegistryObject<Block> ROCK_BLOCK = BLOCKS.register("rock", () -> new Block(BlockBehaviour.Properties.of(Material.STONE)));
+public static final RegistryObject<Block> ROCK_BLOCK = BLOCKS.register("rock", () -> new Block(BlockBehaviour.Properties.of()));
```
Comment (Matyrobbrt) [inline docs/concepts/registries.md]:
```suggestion
public static final RegistryObject<Block> ROCK_BLOCK = BLOCKS.register("rock", () -> new Block(BlockBehaviour.Properties.of().mapColor(MapColor.STONE)));
```

File: docs/items/index.md [hunk 2]
```diff
@@ -45,25 +45,23 @@ You can also enable or disable items being added through a `FeatureFlag` in the
...
-    // Set icon of creative tab
-    .icon(() -> new ItemStack(ITEM.get()))
-    // Add default items to tab
-    .displayItems((params, output) -> {
-      output.accept(ITEM.get());
-      output.accept(BLOCK.get());
-    })
-  );
-}
+public static final RegistryObject<CreativeModeTab> EXAMPLE_TAB = REGISTRAR.register("example", () -> CreativeModeTab.Builder.of()
```
Comment (Matyrobbrt) [inline docs/items/index.md]:
```suggestion
public static final RegistryObject<CreativeModeTab> EXAMPLE_TAB = REGISTRAR.register("example", () -> CreativeModeTab.builder()
```

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

File: docs/items/index.md [hunk 3]
```diff
@@ -45,25 +45,23 @@ You can also enable or disable items being added through a `FeatureFlag` in the
 
 ### Custom Creative Tabs
 
-A custom `CreativeModeTab` can be created via `CreativeModeTabEvent$Register#registerCreativeModeTab` on the [mod event bus][modbus]. This takes in the name of the tab and a consumer of the builder. In addition, a list of `ResourceLocation`s or `CreativeModeTab`s can be provided to determine where this tab should be located.
+A custom `CreativeModeTab` must be [registered][registering]. The builder can be created via `CreativeModeTab$Builder#of`. The tab can set the title, icon, default items, and a number of other properties. In addition, Forge provides additional methods to customize the tab's image, label and slot colors, where the tab should be ordered, etc.
```
Comment (Matyrobbrt) [inline docs/items/index.md]:
```suggestion
A custom `CreativeModeTab` must be [registered][registering]. The builder can be created via `CreativeModeTab#builder`. The tab can set the title, icon, default items, and a number of other properties. In addition, Forge provides additional methods to customize the tab's image, label and slot colors, where the tab should be ordered, etc.
```

File: docs/resources/server/loottables.md [hunk 4]
```diff
@@ -11,13 +11,13 @@ Most loot tables within vanilla are data driven via JSON. This means that a mod
...
+A loot table is referenced by its `ResourceLocation` which points to `data/<namespace>/loot_tables/<path>.json`. The `LootTable` associated with the reference can be obtained using `LootDataResolver#getLootTable`, where `LootDataResolver` can be obtained via `MinecraftServer#getLootData`.
 
-A loot table is always generated within a given context. The `LootContext` defines the level the table is generated in, a specific randomizer and seed if desired, luck for better generation, the `LootContextParam`s which define scenario context, and any dynamic information that should occur on activation. A loot context can be created using the constructor for `LootContext$Builder` and built using `LootContext$Builder#create`.
+A loot table is always generated with given parameters. The `LootParams` contains the level the table is generated in, luck for better generation, the `LootContextParam`s which define scenario context, and any dynamic information that should occur on activation. The `LootParams` can be created using the builder constructor `LootParams$Builder` and built via `LootParams$Builder#create` by passing in the `LootContextParamSet`.
```
Comment (Matyrobbrt) [inline docs/resources/server/loottables.md]:
```suggestion
A loot table is always generated with given parameters. The `LootParams` contains the level the table is generated in, luck for better generation, the `LootContextParam`s which define scenario context, and any dynamic information that should occur on activation. The `LootParams` can be created using the constructor of the `LootParams$Builder` builder, and built via `LootParams$Builder#create` by passing in the `LootContextParamSet`.
```

File: docs/gui/screens.md [hunk 5]
```diff
@@ -262,15 +261,15 @@ Starting with `#render`, the most common override (and typically the only case)
...
-public void render(PoseStack pose, int mouseX, int mouseY, float partialTick) {
-    this.renderBackground(pose);
-    super.render(pose, mouseX, mouseY, partialTick);
+public void render(GuiGraphics graphics, int mouseX, int mouseY, float partialTick) {
+    this.renderBackground(graphics);
+    super.render(graphics, mouseX, mouseY, partialTick);
 
     /*
      * This method is added by the container screen to render
      * a tooltip for whatever slot is hovered over.
```
Comment (Matyrobbrt) [inline docs/gui/screens.md]:
```suggestion
     * the tooltip of the hovered slot.
```

File: docs/gui/screens.md [hunk 6]
```diff
@@ -13,9 +13,9 @@ Information on how to relativize your coordinates will be within the [screen] se
...
+## Gui Graphics
 
-Any GUI rendered by Minecraft is an instance of a `GuiComponent`. `GuiComponent`s contain the most basic methods used to render the most commonly used objects. These fall into three categories: colored rectangles, strings, and textures. There is also an additional method for rendering a snippet of a component (`#enableScissor` / `#disableScissor`). All methods take in a `PoseStack` which applies the transformations necessary to properly render where the component should be rendered. Additionally, colors are in the [ARGB][argb] format.
+Any GUI rendered by Minecraft is typically done using `GuiGraphics`. `GuiGraphics` is the first parameter to almost all rendering methods; it contains basic methods to render commonly used objects. These fall into five categories: colored rectangles, strings, and textures, items, and tooltips. There is also an additional method for rendering a snippet of a component (`#enableScissor` / `#disableScissor`). `GuiGraphics`. `GuiGraphics` also exposes the `PoseStack` which applies the transformations necessary to properly render where the component should be rendered. Additionally, colors are in the [ARGB][argb] format.
```
Comment (Matyrobbrt) [inline docs/gui/screens.md]:
```suggestion
Any GUI rendered by Minecraft is typically done using `GuiGraphics`. `GuiGraphics` is the first parameter to almost all rendering methods; it contains basic methods to render commonly used objects. These fall into five categories: colored rectangles, strings, and textures, items, and tooltips. There is also an additional method for rendering a snippet of a component (`#enableScissor` / `#disableScissor`). `GuiGraphics` also exposes the `PoseStack` which applies the transformations necessary to properly render where the component should be rendered. Additionally, colors are in the [ARGB][argb] format.
```

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

File: docs/gettingstarted/index.md [hunk 7]
```diff
@@ -67,7 +67,7 @@ base.archivesName = 'mymod'
 
     ```gradle
     // In some build.gradle
-    base.archivesName = 'mymod'
+    archivesBaseName = 'mymod'
```
Comment (Matyrobbrt) [inline docs/gettingstarted/index.md]: Why did you change this?

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

File: docs/gettingstarted/index.md (hunk 7 above)
Comment (ChampionAsh5357) [reply docs/gettingstarted/index.md]: Because it was a mistake. It's within a note saying that the Forge MDK was using `archivesBaseName`. However, since we're no longer doing that. I can just remove the entire note itself.

**Review by Matyrobbrt — APPROVED** [review]

File: docs/concepts/lifecycle.md [hunk 8]
```diff
@@ -30,10 +30,12 @@ public class MyMod {
 Registry Events
 ---------------
 
-The registry events are fired after the mod instance construction. There are two: `NewRegistryEvent` and `RegisterEvent`. These events are fired synchronously during mod loading.
+The registry events are fired after the mod instance construction. There are two: `NewRegistryEvent`, `DataPackRegistryEvent$NewRegistry` and `RegisterEvent`. These events are fired synchronously during mod loading.
```
Comment (Alpvax) [inline docs/concepts/lifecycle.md:33]: Minor thing missed: There are now 3 events, not 2
